include/mgba-util/socket.h

```diff
@@ -438,6 +441,56 @@ static inline int SocketPoll(size_t nSockets, Socket* reads, Socket* writes, Soc
 	return result;
 }
 
+static inline int SocketResolveHost(const char* addrString, struct Address* destAddress) {
+	int result = 0;
+#ifdef USE_GETHOSTBYNAME
+	// TODO: This isn't threadsafe. Fix!
+#ifdef GEKKO
+	struct hostent* host = net_gethostbyname(addrString);
+#else
+	struct hostent* host = gethostbyname(addrString);
+#endif
+	if (!host) {
+		result = errno;
+		goto error;
```

Comment:
Since `error` is the end of the functionin this case, an early return is preferred.

Reply:
It's not the end of the function. It calls `freeaddrinfo` first.

Reply:
Oh wait. Wrong ifdef branch.

Huh. I thought I removed that one already. Must have rolled back.